### ISSUE TITLE
Bot Emulator stuck in "sending" state

### ISSUE THREAD

**Issue by cj2342** [Microsoft/BotFramework-Emulator#674]
Hi,

I'm following the example in the following tutorial:

https://mva.microsoft.com/en-US/training-courses/creating-bots-in-the-microsoft-bot-framework-using-nodejs-16759?l=Pfzej8HyC_4004668937

As far as I can see the script I have is exactly the same but when I run the emulator and send a message it never fully sends and is stuck "sending". I don't see anything in the log either (which could mean I haven't connected perhaps?)

Hopefully I'm missing something simple. Any ideas?

**Comment by jwiley84** [comment]
Hi CJ. I'm reaching out to the author of the tutorial for some information. In the mean time, can you go ahead and post your code so we can take a look at it and maybe spot the problem?

**Comment by cj2342** [comment]
Hi,

Thanks for the reply.

As I was going to copy my code in I noticed that the part where I was declaring a reference to restify had a small underline. It mentioned that it "could not find a declaration file" and to try "npm install @types/restify".

I thought I'd done that previously but apparently hadn't. I opened up Terminal and did it again, and now I can connect fine.

Thanks again and apologies,
